server/src/test/java/io/crate/analyze/AlterTableAddColumnAnalyzerTest.java

```diff
@@ -192,4 +192,16 @@ public void add_multiple_columns_adding_same_name_primitive_throws_an_exception(
             .isExactlyInstanceOf(IllegalArgumentException.class)
             .hasMessageContaining("column \"col\" specified more than once");
     }
+
+    @Test
+    public void test_check_constraint_cannot_be_added_to_nested_object_sub_column() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("create table t (i int, o object)")
+            .build();
+
+        assertThatThrownBy(
+            () -> analyze("alter table t add column o1 object as (o2 object as (b int check (b > 100)))"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Constraints on nested columns are not allowed");
```

Comment:
Can we have a TEXT case to ensure that it's not "accidentally" supported?

Reply:
thx, done.